OHS-Hosting-Infrastructure/complaint-tracker
# Pagination backend

## Description of change

Adds methods to calculate offset and limit from page request to complaints list
Creates default parameters in ApiRequest so that subclasses (like Hses::Issues) need not worry about them,
and instead use `parameters` method to provide their own options

Note: the functionality of `query` was moved to `parameters` and `query` now combines default parameters and subclass parameters

Another note: in doing this, I realized that when HSES changed the API spec to use `form` and removed `deepObject`, we seem to have lost the URL params we were hoping for, namely: we wanted `page[offset]=x&page[limit]=x` but the current API wants `offset=x&limit=x`. This is something we may want to revisit.

## Acceptance Criteria

- Passes limit / offset information to the HSES API

## How to test

~Well, manually you can drop `page=2` or whatever you fancy into the `/complaints` URL and see the results change.~ (removed after comment)

## Issue(s)

* closes https://github.com/OHS-Hosting-Infrastructure/complaint-tracker/issues/134


## Checklist

To be completed by the submitter:


- [?] Code is meaningfully tested
- ~[] Meets accessibility standards (WCAG 2.1 Levels A, AA)~
- ~[] API Documentation updated~

## To the Reviewer

This project is using [Conventional Comments](https://conventionalcomments.org/) to give structure
and context to PR comments. Please use these labels in your comments.

* **praise:**
* **nitpick:**
* **suggestion:**
* **issue:**
* **question:**
* **thought:**
* **chore:**
